crypto/core_fetch.c

```diff
@@ -31,6 +31,20 @@ static void ossl_method_construct_this(OSSL_PROVIDER *provider,
     struct construct_data_st *data = cbdata;
     void *method = NULL;
 
+    /*
+     * Check to see if a method has already been constructed from this
+     * implementation.  Note that we use the property definition as a
+     * property query, which stresses that they MUST have compatible
+     * syntax, or else...  (in other words, it must always be possible
+     * to copy a property definition and paste that wherever a property
+     * is being queried.
+     */
+    if ((method = data->mcm->get(data->libctx, NULL, data->operation_id,
```

Comment:
I kind of wonder, if you should also try to `data->mcm->get(data->libctx, data->store, ...)` here,
especially when the put to NULL-store is optional?

Reply:
Why?  `data->store` is a temporary store, so that would really only be useful if the provider has more than one implementation with the same \< name, properties \> pair.  This *might* of course be an issue, but in that case, I would prefer to fix that in a separate PR.

Reply:
I really don't know this code good enough, I am just wondering.
Do you try to get the method from a global store even when this
the constructor is not going to add it to the global store `!((data->force_store || !no_store)`

Reply:
You do have a point that it's pretty pointless to look for it in the global store in that case.  Ok, will fix

Reply:
I kind of wonder if this sequence needs a lock, especially when duplicates
in the store create a memory leak ?

Reply:
why was there a memory leak in the first place, I mean isn't the put method
"expected" to do the right thing ?